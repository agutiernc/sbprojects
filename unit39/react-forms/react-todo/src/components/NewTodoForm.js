import React, { useState } from 'react';
import { v4 as uuidv4 } from 'uuid';

const NewTodoForm = ({ newTodo }) => {
  const initialState = { todo: '' }

  const [formData, setFormData] = useState(initialState)

  const handleChange = (e) => {
    const { name, value } = e.target

    setFormData(data => ({
      ...data,
      [name]: value
    }))
  }

  const handleSubmit = (e) => {
    e.preventDefault()

    newTodo({ ...formData, id: uuidv4() })
    
    setFormData(initialState)
  }

  return (
    <div>
      <form onSubmit={handleSubmit}>
        <label htmlFor="todo">Add todo:</label>
        <input
          id="todo"
          type="text"
          name="todo"
          value={formData.todo}
          onChange={handleChange}
        />

        <button>Submit</button>
      </form>
    </div>
  )
}

export default NewTodoForm;