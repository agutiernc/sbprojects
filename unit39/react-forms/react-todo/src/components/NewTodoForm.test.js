import React from "react";
import { render } from "@testing-library/react";
import NewTodoForm from "./NewTodoForm";

// smoke test
it('renders without crashing', () => {
  render(<NewTodoForm />)
})

// snapshot test
it('matches snapshot', () => {
  const { asFragment } = render(<NewTodoForm />)

  expect(asFragment()).toMatchSnapshot()
})