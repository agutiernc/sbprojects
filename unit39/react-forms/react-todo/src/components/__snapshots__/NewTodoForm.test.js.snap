// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <form>
      <input
        id="todo"
        name="todo"
        placeholder="Enter todo..."
        type="text"
        value=""
      />
      <button>
        Add todo
      </button>
    </form>
  </div>
</DocumentFragment>
`;
